e2e/single-cluster/delete_namespaces_test.go

```diff
@@ -0,0 +1,107 @@
+package singlecluster_test
+
+import (
+	"errors"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher/fleet/e2e/testenv"
+	"github.com/rancher/fleet/e2e/testenv/kubectl"
+)
+
+var _ = Describe("delete namespaces", func() {
+	var (
+		k               kubectl.Command
+		targetNamespace string
+		deleteNamespace bool
+	)
+
+	type TemplateData struct {
+		TargetNamespace string
+		DeleteNamespace bool
+	}
+
+	BeforeEach(func() {
+		k = env.Kubectl.Namespace(env.Namespace)
+		deleteNamespace = false
+
+		DeferCleanup(func() {
+			_, _ = k.Delete("ns", "my-custom-namespace", "--wait=false")
+		})
+	})
+
+	JustBeforeEach(func() {
+		err := testenv.ApplyTemplate(k, testenv.AssetPath("single-cluster/delete-namespace/gitrepo.yaml"),
+			TemplateData{targetNamespace, deleteNamespace})
+
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			out, err := k.Namespace(targetNamespace).Get("configmaps")
+			if err != nil {
+				return err
+			}
+
+			if !strings.Contains(out, "app-config") {
+				return errors.New("expected configmap is not found")
+			}
+
+			return nil
+		}).ShouldNot(HaveOccurred())
+	})
+
+	When("delete namespaces is false", func() {
+		BeforeEach(func() {
+			targetNamespace = "my-custom-namespace"
+		})
+
+		It("targetNamespace is preserved when GitRepo is deleted", func() {
+			out, err := k.Delete("gitrepo", "my-gitrepo", "-n", "fleet-local")
+			Expect(err).ToNot(HaveOccurred(), out)
+
+			Eventually(func() string {
+				out, _ := k.Get("namespaces", targetNamespace)
```

Comment:
Precisely: since we want to check the _existence_ of the namespace, we need to ensure that our tests do precisely that.
With the current code here, if `k.Get("namespaces", targetNamespace)` returns an error with `out` being `Error from server (NotFound): namespaces "<target-namespace>" not found`, which could happen if the namespace had been deleted, our test case would not detect this.
We need to check for the error to make sure that we catch possible regressions.

Reply:
Ah I see. Yes, that makes sense. It's been updated. Thank you! :)